### ISSUE TITLE
Training loop is non-learning: policies reset every episode, motor outputs collapse

### ISSUE THREAD

**Issue by jimlbeaver** [jimlbeaver/Bug-Sim#16]
﻿## Summary

Diagnostic of the in-flight `exp_001` run (gen ~100/1000, `--mode visual --episodes 1000`) shows that **no learning is accumulating across episodes** and **agents barely actuate their motors**. The simulation process advances, but the system is stuck at a degenerate "do nothing" fixed point that it re-discovers from scratch every episode.

This issue documents the root causes. A separate design doc (`docs/REMEDIATION_PLAN_training_loop.md`) and follow-up PRs will fix them.

## Symptoms

- Bugs do not appear to move in the MuJoCo viewer.
- Every per-species console line reads `mean=0.00, min=0.00, max=0.00` regardless of generation.
- Every `run_metadata.json` has `fitness_score: 0.0`.
- Episodes terminate after ~1000-1500 steps (energy decays to 0), far below `simulation.max_timesteps = 10000`.

## Evidence

From `scripts/diagnose_training.py` over 140 saved `carnivore_0` checkpoints (gen 0 → gen 99):

```
Wheel-output stats over 512 random observations (action[0:2] after tanh)
  fresh   wheel  min=-0.96  max=+0.94  mean=+0.00  std=+0.48
  first   wheel  min=-0.49  max=+0.45  mean=-0.00  std=+0.16
  middle  wheel  min=-0.47  max=+0.38  mean=-0.01  std=+0.14
  last    wheel  min=-0.46  max=+0.35  mean=-0.00  std=+0.12

L2 distance: last (gen 99) vs first (gen 0) checkpoint
  shared_layers.0.weight   ||delta||=  7.72
  shared_layers.2.weight   ||delta||= 10.28
  policy_head.weight       ||delta||=  2.71
  value_head.weight        ||delta||=  1.01
```

Saved checkpoints from gen 0, gen 49, and gen 99 all sit roughly the same L2 distance from a fresh Xavier init — consistent with each one being "random init → ~7 PPO updates that collapse outputs". There is no upward trajectory in policy quality across generations.

### 1. No policy persistence between episodes (CRITICAL)

`bug_world/simulation/environment.py:180-201` rebuilds every `BugAgentImpl` with a fresh `NeuralPolicyImpl(...)` on each `reset()`. The runner never calls `policy.load(...)` from the previous gen's checkpoint, nor does it reuse the previous episode's bug objects.

Effect: every episode is gen-0 again. The 1000-episode run is 1000 independent re-rolls of the same fresh→squashed trajectory.

### 2. Reward function is policy-agnostic

`environment.py:492-495`:
```python
reward = _BASE_SURVIVAL_REWARD + _ENERGY_REWARD_SCALE * stats.energy
      = -0.01 + 0.001 * energy
```

Energy decays at `-0.1/step` regardless of action. The only action-dependent reward term comes indirectly via predation/food contacts, which are sparse. PPO sees almost no gradient distinguishing "move" from "don't move", and entropy regularization pulls the policy mean toward zero.

### 3. PPO is off-policy

Rollouts use deterministic `policy.forward()` (`tanh(mean)` only), but `get_action_distribution` constructs `Normal(mean, sigma=0.5)` to compute `old_log_probs` and the PPO ratio. The actions in the replay buffer were never sampled from that distribution, so the PPO objective is computing ratios against the wrong reference and gradients are noisy.

### 4. FedAvg compresses policy norms

`config.federated.aggregate_every_n_steps = 500` averages all 5 carnivore (etc.) policy tensors twice per ~1000-step episode. Averaging N i.i.d. near-zero-mean tensors shrinks the norm by `~1/sqrt(N)`, so policies are pushed even closer to zero outputs.

### 5. Position tracking is broken (BUG)

`bug_world/agents/agent.py:131-133` returns `position_x=0.0, position_y=0.0` unconditionally in `get_stats()`. As a result, even if `FitnessEvaluatorImpl` were wired in (it currently isn't, see #6), `distance_traveled` would always be 0.

### 6. Fitness module is not wired into the run loop

`FitnessEvaluatorImpl` is constructed by `SimulationRunner.__init__` but never called in `run()`. Both the runner's `_log_episode_summary` and the CLI use their own ad-hoc reward aggregations.

### 7. Display bug: `fitness_score: 0.0` everywhere

`bug_world/__main__.py:132-134`:
```python
sum(getattr(s, "reward", 0.0) for s in trajectory)
```

`AgentStats` has no `reward` field (see `bug_world/agents/api.py:25`). `getattr(..., "reward", 0.0)` always returns 0.0. That number is then stored as `fitness_score` in every `run_metadata.json`.

## Affected Modules

- `simulation` — `runner.py`, `environment.py` (load/save policies across episodes; wire fitness; stochastic rollout)
- `neural` — `ppo.py`, `policy.py` (expose stochastic `act()` for rollout; align log_probs with rollout distribution)
- `agents` — `agent.py` (real `position_x/y` from MuJoCo; correct reward consumption)
- `__main__.py` — fix reward aggregation; use `FitnessEvaluator`
- `fitness` — verify scoring path now that positions and rewards are real
- `federated` — make aggregation cadence configurable per-episode or disable for warmup
- `config` — add `simulation.persist_policies_across_episodes`, `neural.ppo.exploration_std`, `neural.rollout_mode` ("deterministic" | "stochastic")

## Out of scope for this issue

- Reward shaping research (will be a follow-up; this issue only restores a working learning signal)
- Replacing PPO with anything else (Phase 2)

## Acceptance criteria

1. A 10-episode run shows `||policy_weights(gen N) - policy_weights(gen N-1)||` smaller than `||policy_weights(gen N) - fresh_init||` for every N>=1. (Policies actually persist.)
2. Wheel-output std over random observations grows or stays >=0.3 across an episode (no collapse).
3. `fitness_score` in `run_metadata.json` is non-zero and varies between agents.
4. `position_x`/`position_y` in `AgentStats` reflect actual MuJoCo body positions.
5. Visual mode shows bugs translating around the arena within the first episode.
6. All existing tests pass; new regression tests cover items 1-4.

## Diagnostic artifacts

- `scripts/diagnose_training.py` — reproduces the wheel-collapse and weight-drift analysis
- `scripts/check_run_status.ps1` — periodic status snapshot of the active run
